React/Base/RCTUtils.m

```diff
@@ -337,6 +337,42 @@ BOOL RCTRunningInTestEnvironment(void)
   return isTestEnvironment;
 }
 
+BOOL RCTRunningInAppExtension(void)
+{
+  return [[[[NSBundle mainBundle] bundlePath] pathExtension] isEqualToString:@"appex"];
+}
+
+id RCTSharedApplication(void)
+{
+  if (RCTRunningInAppExtension()) {
+    return nil;
+  }
+  
+  return [[UIApplication class] performSelector:@selector(sharedApplication)];
```

Comment:
I think the `performSelector` is necessary.  If we try to invoke the method directly, the static analysis will catch us at compile time.  We know it's safe to call, since we've done the runtime check, but Xcode will still yell at us.